### PR TITLE
Adding support for JIT Fusion on Windows for CUDA

### PR DESCRIPTION
**Pull request by gavrielstate**
This pull request adds the necessary Windows DLL code to be able to support JIT fusion for CUDA. CPU JIT Fusion isn't supported. This also adds all the non-CPU JIT tests back in on Windows.

### CODE REVIEW

**Comment by gavrielstate** [comment]
@peterjc123 - Per suggestion from others, tagging you here to request that you review this.

**Review by peterjc123 — APPROVED** [review]
LGTM.

File: torch/csrc/jit/fuser/cpu/dynamic_library_win.cpp [hunk 1]
```diff
@@ -7,16 +8,29 @@ namespace jit {
 namespace fuser {
 namespace cpu {
 
+
 DynamicLibrary::DynamicLibrary(const char* name) {
   // NOLINTNEXTLINE(hicpp-signed-bitwise)
-  AT_ERROR("NYI: DynamicLibrary on Windows");
+  HMODULE theModule = LoadLibraryA(name);
+  if (theModule)
```
Comment (zdevito) [inline torch/csrc/jit/fuser/cpu/dynamic_library_win.cpp]: Can you add braces for these if statements to match the style of other code?

File: torch/csrc/jit/fuser/cuda/thnvrtc.h [hunk 2]
```diff
@@ -30,4 +30,8 @@ extern "C" typedef struct THNVRTC {
 #undef CREATE_MEMBER
 } THNVRTC;
 
-extern "C" THNVRTC* torch_load_nvrtc();
+extern "C" 
+#ifdef _WIN32
```
Comment (zdevito) [inline torch/csrc/jit/fuser/cuda/thnvrtc.h]: Can you use  the macro "TORCH_API" to do these exports?

**Review by zdevito — COMMENTED** [review]
Thanks! Just a couple nits.

File: torch/csrc/jit/fuser/cpu/dynamic_library_win.cpp (hunk 1 above)
Comment (gavrielstate) [reply torch/csrc/jit/fuser/cpu/dynamic_library_win.cpp]: Done

File: torch/csrc/jit/fuser/cuda/thnvrtc.h (hunk 2 above)
Comment (gavrielstate) [reply torch/csrc/jit/fuser/cuda/thnvrtc.h]: Done.

**Review by apaszke — APPROVED** [review]
Don't those changes enable the CPU fuser as well? IIRC The `DynamicLibrary` stuff is only used in that case, so it might be enough.

**Comment by gavrielstate** [comment]
@apaszke - the CPU Fuser isn't enabled on Windows, since there isn't a good way to find out whether there is an external compiler installed on the system. The dynamiclibrary code is still needed to support loading the thnvrtc wrapper dynamically.

**Comment by xsacha** [comment]
I downloaded the nightly but the CMake doesn't know about thnvrtc. Probably need to add it to TORCH_CUDA_LIBRARIES

**Comment by peterjc123** [comment]
@xsacha I'm fixing it.

**Comment by gavrielstate** [comment]
That's odd. I got the Nightly build from Conda on Saturday and it had the thnvrtc.dll in there. Is this for some other type of packaging?

**Comment by xsacha** [comment]
@gavrielstate Yes, the DLL is there but the caffe2/torch Config's do not reference it.
Just got fixed by @peterjc123 and pushed 9 hours ago.

**Pull request closed without merging**